lib/LanguagePicker.js

```diff
@@ -19,50 +42,83 @@ function LanguagePicker(lang = 'en', config = {}) {
   // Use the given language as first choice
   fallbacks = [lang].concat(fallbacks);
 
-  // Fallback to English
-  if (fallbacks.indexOf('en') === -1) {
-    fallbacks.push('en');
-  }
-
-  // Fallback to nameTag
-  if (this.nameTag) {
-    fallbacks.push(this.nameTag);
-  }
+  // Remove duplicates
+  fallbacks = fallbacks.filter((item, i) => fallbacks.indexOf(item) === i);
 
   // Add prefix to all languages if exists
   // eslint-disable-next-line arrow-body-style
   fallbacks = fallbacks.map((code) => {
     return code === this.nameTag ? code : this.prefix + code;
   });
 
+  // Store initial fallbacks
   this.fallbacks = fallbacks;
 }
 
 LanguagePicker.prototype.newProcessor = function newProcessor() {
-  const { fallbacks, nameTag, prefix } = this;
-  let bestChoiceLang = nameTag || (prefix ? `${prefix}en` : 'en');
-  let bestChoiceValue;
   let firstChoiceValue;
 
   return {
-    addValue(lang, value) {
-      const langIndex = fallbacks.indexOf(lang);
-      if (
-        langIndex > -1 &&
-        langIndex <= fallbacks.indexOf(bestChoiceLang)
-      ) {
-        // Given language is a better fallback
-        bestChoiceLang = lang;
-        bestChoiceValue = value;
+    addValue: (lang, value) => {
+      this.values[lang] = value;
+
+      if (!firstChoiceValue) {
+        firstChoiceValue = value;
+      }
+    },
+    getResult: () => {
+      const valueLangCodes = Object.keys(this.values);
+
+      // Get the best value from the best language fallback:
+      // 1. Requested language
+      if (this.values[this.userLang]) {
+        return this.values[this.userLang];
+      }
+
+      // 2. Fallback language from fallbacks.json
+      for (let i = 0; i < this.fallbacks.length; i++) {
+        if (this.values[this.fallbacks[i]]) {
+          return this.values[this.fallbacks[i]];
```

Comment:
same thing, and use [for-of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop for readability and to avoid hidden bugs.